WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -315,12 +315,19 @@ private bool SanityCheck(IEnumerable<TxOut> expectedOutputs, Transaction unsigne
 		return coinJoinOutputs.IsSuperSetOf(expectedOutputTuples);
 	}
 
-	private async Task SignTransactionAsync(IEnumerable<AliceClient> aliceClients, Transaction unsignedCoinJoinTransaction, RoundState roundState, CancellationToken cancellationToken)
+	private async Task SignTransactionAsync(IEnumerable<AliceClient> aliceClients, Transaction unsignedCoinJoinTransaction, DateTimeOffset signingEndTime, CancellationToken cancellationToken)
 	{
-		foreach (var aliceClient in aliceClients)
-		{
-			await aliceClient.SignTransactionAsync(unsignedCoinJoinTransaction, KeyChain, cancellationToken).ConfigureAwait(false);
-		}
+		var remainingTime = signingEndTime - DateTimeOffset.UtcNow;
+		var delays = remainingTime.SamplePoissonDelays(aliceClients.Count());
+
+		var tasks = Enumerable.Zip(aliceClients, delays,
+			async (aliceClient, delay) =>
```

Comment:
Here, as well as in 1222d0b829967bff9aa5e31535819e6d01765c4d, the delays are computed but never applied. Basically there no `await Task.Delay(delay, cancellationToken)` anywhere.

Reply:
Added